Tests/ASTextNode2SnapshotTests.mm

```diff
@@ -45,13 +120,86 @@ - (void)testTextContainerInset_ASTextNode2
+
+  NSMutableAttributedString *refString = [[NSMutableAttributedString alloc] initWithString:@"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
+          attributes:@{ NSFontAttributeName : [UIFont systemFontOfSize:18.0f] }];
+  NSMutableAttributedString *asString = [refString mutableCopy];
+
+  reference.attributedText = refString;
+  textNode.attributedText = asString;
+
+  CGSize size = (CGSize) {container.bounds.size.width, 120.0};
+  CGPoint origin = (CGPoint) {CGRectGetWidth(container.bounds) / 2 - size.width / 2, CGRectGetHeight(container.bounds) / 2 - size.height / 2}; // center
+
+  textNode.frame = (CGRect) {origin, size};
+  reference.frame = CGRectOffset(textNode.frame, 0, -160.0f);
+
+  textNodeLabel.bounds = (CGRect) {CGPointZero, (CGSize) {container.bounds.size.width, textNodeLabel.font.lineHeight}};
+  origin = (CGPoint) {textNode.frame.origin.x, textNode.frame.origin.y - textNodeLabel.bounds.size.height};
+  textNodeLabel.frame = (CGRect) {origin, textNodeLabel.bounds.size};
+
+  uiLabelLabel.bounds = (CGRect) {CGPointZero, (CGSize) {container.bounds.size.width, uiLabelLabel.font.lineHeight}};
+  origin = (CGPoint) {reference.frame.origin.x, reference.frame.origin.y - uiLabelLabel.bounds.size.height};
+  uiLabelLabel.frame = (CGRect) {origin, uiLabelLabel.bounds.size};
+
+  uiLabelLabel.bounds = (CGRect) {CGPointZero, (CGSize) {container.bounds.size.width, uiLabelLabel.font.lineHeight}};
+  origin = (CGPoint) {textNode.frame.origin.x, textNode.frame.origin.y - uiLabelLabel.bounds.size.height};
+  uiLabelLabel.frame = (CGRect) {origin, uiLabelLabel.bounds.size};
+
+  uiLabelLabel.bounds = (CGRect) {CGPointZero, (CGSize) {container.bounds.size.width, uiLabelLabel.font.lineHeight}};
+  origin = (CGPoint) {reference.frame.origin.x, reference.frame.origin.y - uiLabelLabel.bounds.size.height};
+  uiLabelLabel.frame = (CGRect) {origin, uiLabelLabel.bounds.size};
+
+  description.bounds = textNode.bounds;
+  description.frame = (CGRect) {(CGPoint) {0, container.bounds.size.height * 0.8}, description.bounds.size};
+
+  [container addSubview:reference];
+  [container addSubview:textNode.view];
+  [container addSubview:textNodeLabel];
+  [container addSubview:uiLabelLabel];
+  [container addSubview:description];
+
+  NSArray<LineBreakConfig *> *c = [LineBreakConfig configs];
+  for (LineBreakConfig *config in c) {
+    reference.lineBreakMode = textNode.truncationMode = config.lineBreakMode;
+    reference.numberOfLines = textNode.maximumNumberOfLines = config.numberOfLines;
+    description.text = config.description;
+    [container setNeedsLayout];
+    NSString *identifier = [NSString stringWithFormat:@"%lu_%lu", (unsigned long)config.lineBreakMode, (unsigned long)config.numberOfLines];
```

Comment:
Instead of using the line break mode int value, maybe it’d be easier to understand if we use a string representation of the mode here?

Reply:
okay